src/helpers/figures/charts/runtime/chart_data_extractor.ts

```diff
@@ -402,10 +402,18 @@ function interpolateData(
         )[0];
       }
       case "logarithmic": {
+        const logLabels: number[] = [];
+        const filteredValues: number[] = [];
+        for (let i = 0; i < values.length; i++) {
+          if (normalizedLabels[i] > 0) {
+            logLabels.push(values[i]);
+            filteredValues.push(Math.log(normalizedLabels[i]));
+          }
+        }
         return predictLinearValues(
-          [values],
-          logM([normalizedLabels]),
-          logM([normalizedNewLabels]),
+          [logLabels],
+          [filteredValues],
```

Comment:
```suggestion
          [logM(filteredValues)],
```
you can keep the current call to logM, iI think it's more readable like this instead of having another point where we apply logarithms